### PR TITLE
[DC-1146] Drop duplicate COPE response rows

### PR DESCRIPTION
**Pull request by ksdkalluri**
* Adds Drop duplicate COPE response rows cleaning rule
* Adds Unit tests and integration tests

Signed-off-by: Krishna Kalluri <krishnasaikalluri@gmail.com>

### CODE REVIEW

File: tests/unit_tests/data_steward/cdr_cleaner/cleaning_rules/drop_cope_duplicate_responses_test.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    def test_get_query_specs(self):
+        # Pre conditions
+        self.assertEqual(self.rule_instance.affected_datasets,
+                         [clean_consts.RDR])
+
+        # Test
+        results_list = self.rule_instance.get_query_specs()
+        # Post conditions
+        sandbox_query = dict()
+        sandbox_query[
```
Comment (ChaoPang) [inline tests/unit_tests/data_steward/cdr_cleaner/cleaning_rules/drop_cope_duplicate_responses_test.py:44]: Please update this test to generate queries in a different way other than calling the functions

File: tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/drop_cope_duplicate_responses_test.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+        Creates common expected parameter types from cleaned tables and a common
+        fully qualified (fq) dataset name string to load the data.
+        """
+        fq_dataset_name = self.fq_table_names[0].split('.')
+        self.fq_dataset_name = '.'.join(fq_dataset_name[:-1])
+
+        super().setUp()
+
+    def test_field_cleaning(self):
```
Comment (ChaoPang) [inline tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/drop_cope_duplicate_responses_test.py:81]: Use a more informative name perhaps?

File: data_steward/cdr_cleaner/cleaning_rules/drop_cope_duplicate_responses.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    (
+  SELECT
+  observation_id
+FROM (
+  SELECT
+    *,
+    DENSE_RANK() OVER(PARTITION BY person_id,
+     observation_source_concept_id,
+     observation_source_value,
+     cope_month
```
Comment (ChaoPang) [inline data_steward/cdr_cleaner/cleaning_rules/drop_cope_duplicate_responses.py:63]: I think we need to add `value_source_value` to the partition

File: tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/drop_cope_duplicate_responses_test.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    person_id,
+    observation_concept_id,
+    observation_date,
+    observation_datetime,
+    observation_type_concept_id,
+    observation_source_concept_id,
+    observation_source_value,
+    value_source_value,
+    questionnaire_response_id)
+VALUES
```
Comment (ChaoPang) [inline tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/drop_cope_duplicate_responses_test.py:101]: Add the following test cases
- Exactly the same column values except for `observation_id`, we want to drop this
- Different `observation_id` and different `value_source_value`, keep everything else the same, we want to preserve this record.

**Review by ChaoPang — CHANGES_REQUESTED** [review]
Good work! Please address the issues raised in the comments

File: data_steward/cdr_cleaner/cleaning_rules/drop_cope_duplicate_responses.py [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+
+    def get_delete_statement(self, project_id, dataset_id,
+                             survey_version_table):
+        duplicate_id_query = IDENTIFY_DUPLICATE_COPE_RESPONSES_TEMPLATE.render(
+            project=project_id,
+            dataset=dataset_id,
+            cope_survey_version_table=survey_version_table)
+        delete_duplicates_template = REMOVE_DUPLICATE_COPE_RESPONSES_TEMPLATE.render(
+            project=project_id, dataset=dataset_id)
+        return delete_duplicates_template + duplicate_id_query
```
[the remark this answers is not in the document]
Comment (ksdkalluri) [reply data_steward/cdr_cleaner/cleaning_rules/drop_cope_duplicate_responses.py:144]: updated!

File: tests/unit_tests/data_steward/cdr_cleaner/cleaning_rules/drop_cope_duplicate_responses_test.py (hunk 1 above)
Comment (ksdkalluri) [reply tests/unit_tests/data_steward/cdr_cleaner/cleaning_rules/drop_cope_duplicate_responses_test.py:44]: updated!

File: tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/drop_cope_duplicate_responses_test.py (hunk 2 above)
Comment (ksdkalluri) [reply tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/drop_cope_duplicate_responses_test.py:81]: Updated

File: tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/drop_cope_duplicate_responses_test.py (hunk 4 above)
Comment (ksdkalluri) [reply tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/drop_cope_duplicate_responses_test.py:101]: done added!

**Review by ChaoPang — APPROVED** [review]
Approved. The test data in the integration test has a weird indentation, I don't think it's a big issue though.

**Pull request merged**